pkg/apis/apps/validation/validation.go

```diff
@@ -177,7 +189,9 @@ func ValidateStatefulSetUpdate(statefulSet, oldStatefulSet *apps.StatefulSet, op
 	// thing to do it delete such an instance, but if there is a finalizer, it
 	// would need to pass update validation.  Name can't change anyway.
 	allErrs := apivalidation.ValidateObjectMetaUpdate(&statefulSet.ObjectMeta, &oldStatefulSet.ObjectMeta, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidateStatefulSetSpec(&statefulSet.Spec, field.NewPath("spec"), opts)...)
+	// we need to allow updates to existing StatefulSet
+	setOpts := StatefulSetValidationOptions{AllowDNS1123ServiceName: true}
```

Comment:
```suggestion
	setOpts := StatefulSetValidationOptions{
		AllowInvalidServiceName: true, // serviceName is immutable, tolerate existing invalid names on update
	 }
```

_if_ this was a mutable field, we would have _conditionally_ set AllowInvalidServiceName to something like this:

```go
hasInvalidServiceName := len(oldStatefulSet.Spec.ServiceName) > 0 && len(apivalidation.ValidateDNS1123Label(oldStatefulSet.Spec.ServiceName)) > 0

setOpts := StatefulSetValidationOptions{
  AllowInvalidServiceName: hasInvalidServiceName,
}
```